internal/cli/errors.go

```diff
@@ -0,0 +1,98 @@
+/*
+ Copyright 2024 The Knative Authors
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package cli
+
+import (
+	"io"
+	"regexp"
+	"strings"
+
+	"github.com/fatih/color"
+	"github.com/spf13/cobra"
+	outlogging "knative.dev/client/pkg/output/logging"
+	"knative.dev/client/pkg/output/term"
+	"knative.dev/kn-plugin-event/pkg/errors"
+)
+
+const likelyErrorChainDepth = 3
+
+var (
+	numbersRE = regexp.MustCompile(`\s+\d+\s+`)
+	stringRE  = regexp.MustCompile(`\s+"[^"]+"\s+`)
+)
+
+func errorHandler(err error, cmd *cobra.Command) bool {
```

Comment:
This error handler prints errors like that:
![Screenshot from 2024-10-22 19-42-11](https://github.com/user-attachments/assets/01b5d622-bcaa-4b42-b1f7-d886dbf0a969)

Reply:
Here's an example of successful log file contents: https://gist.github.com/cardil/6151fae7d7b9c5f8ffdcf6469d6ea27b

And here is a failure example: https://gist.github.com/cardil/af129bb83cbfb018b93921babd513467

Reply:
With https://github.com/knative-extensions/kn-plugin-event/pull/375/commits/ff897901d5f42e1e79471f0ff31a8f7b8a330383 I added gathering of K8s' events as well. That should collect all the info we'll need for debugging: job, pods, and events.

Here's updated successful log (converted to JSON array, for readability): https://gist.github.com/cardil/d8b7963c10e7b6e3befcc112916965c7

And here is a failure example (also as JSON array): https://gist.github.com/cardil/d1216a51f483f242ed39517f4ede9a9c

Reply:
Here's how the error handling and the [kn client tui package](https://pkg.go.dev/knative.dev/client/pkg/output/tui) animation plays well:

![kn-event-fancy-err](https://github.com/user-attachments/assets/9576ddfa-9b8f-4017-8435-4253d3142355)